unit_tests/testAndersonAcceleration/testAndersonAcceleration.f90

```diff
@@ -60,539 +55,213 @@ PROGRAM testAndersonAcceleration
-  CALL testAndAcc%X%get(4,tmp)
-  ASSERT(tmp==2.5945_SRK,'%step(1)')
-  CALL testAndAcc%X%get(5,tmp)
-  ASSERT(tmp==3.1245_SRK,'%step(1)')
-
-  CALL testAndAcc%X%set(1,0.980_SRK)
-  CALL testAndAcc%X%set(2,1.580_SRK)
-  CALL testAndAcc%X%set(3,2.012_SRK)
-  CALL testAndAcc%X%set(4,4.141_SRK)
-  CALL testAndAcc%X%set(5,5.198_SRK)
-
-  CALL testAndAcc%step()
-
-  WRITE(*,*) "step(2)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
-  ENDDO
-
-  CALL testAndAcc%X%set(1,0.987_SRK)
-  CALL testAndAcc%X%set(2,1.735_SRK)
-  CALL testAndAcc%X%set(3,2.374_SRK)
-  CALL testAndAcc%X%set(4,4.095_SRK)
-  CALL testAndAcc%X%set(5,5.125_SRK)
-
-  CALL testAndAcc%step()
-
-  WRITE(*,*) "step(3)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
-  ENDDO
-
-  CALL testAndAcc%X%set(1,0.992_SRK)
-  CALL testAndAcc%X%set(2,1.840_SRK)
-  CALL testAndAcc%X%set(3,2.629_SRK)
-  CALL testAndAcc%X%set(4,4.057_SRK)
-  CALL testAndAcc%X%set(5,5.074_SRK)
+  REAL(SRK) :: UnAccErr(10),AccErr(10),mySol(10000),exSol(10000),inSol(10000),R(10000),Norm
 
-  CALL testAndAcc%step()
-
-  WRITE(*,*) "step(4)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
-  ENDDO
-
-  CALL testAndAcc%X%set(1,0.995_SRK)
-  CALL testAndAcc%X%set(2,1.897_SRK)
-  CALL testAndAcc%X%set(3,2.754_SRK)
-  CALL testAndAcc%X%set(4,4.039_SRK)
-  CALL testAndAcc%X%set(5,5.049_SRK)
-
-  CALL testAndAcc%step()
-
-  WRITE(*,*) "step(5)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
-  ENDDO
-
-  CALL testAndAcc%X%set(1,0.998_SRK)
-  CALL testAndAcc%X%set(2,1.948_SRK)
-  CALL testAndAcc%X%set(3,2.876_SRK)
-  CALL testAndAcc%X%set(4,4.020_SRK)
-  CALL testAndAcc%X%set(5,5.025_SRK)
-
-  CALL testAndAcc%step()
-
-  WRITE(*,*) "step(6)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
-  ENDDO
-
-  CALL testAndAcc%X%set(1,0.999_SRK)
-  CALL testAndAcc%X%set(2,1.967_SRK)
-  CALL testAndAcc%X%set(3,2.930_SRK)
-  CALL testAndAcc%X%set(4,4.012_SRK)
-  CALL testAndAcc%X%set(5,5.014_SRK)
-
-  CALL testAndAcc%step()
-
-  WRITE(*,*) "step(7)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
-  ENDDO
-
-  CALL testAndAcc%X%set(1,0.999_SRK)
-  CALL testAndAcc%X%set(2,1.990_SRK)
-  CALL testAndAcc%X%set(3,2.979_SRK)
-  CALL testAndAcc%X%set(4,4.004_SRK)
-  CALL testAndAcc%X%set(5,5.005_SRK)
-
-  CALL testAndAcc%step()
-
-  WRITE(*,*) "step(8)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
-  ENDDO
-
-  CALL testAndAcc%X%set(1,1.000_SRK)
-  CALL testAndAcc%X%set(2,1.993_SRK)
-  CALL testAndAcc%X%set(3,2.990_SRK)
-  CALL testAndAcc%X%set(4,4.002_SRK)
-  CALL testAndAcc%X%set(5,5.001_SRK)
-
-  CALL testAndAcc%step()
+  !Init Anderson
+  CALL testAndAcc%init(mpiTestEnv,optList)
 
-  WRITE(*,*) "step(9)"
-  DO i=1,5
-    CALL testAndAcc%X%get(i,tmp)
-    WRITE(*,*) i, tmp
+  !Exact chosen solution. Operator G will simply be taking a weighted average between
```

Comment:
why make this random?  I'm fine with you hard coding a random solution... but you could literally get a different result every time you run the unit test?  Is anything forcing a constant seed?